web/app.js

```diff
@@ -1235,6 +1244,11 @@ const PDFViewerApplication = {
           return false;
         }
         console.warn("Warning: JavaScript is not supported");
+        this.externalServices.reportTelemetry({
```

Comment:
Is there a reason why this cannot be done (unconditionally) in the `fallback` method? There are more kinds of unsupported features, so why are only these special-cased here?

Reply:
Yeah I originally thought the same thing, but was hesitant given my work in #11953 (which I think will land before this patch). There, we would trigger the fallback for form and script errors only when the user clicks, and `featureId` would be null. 

So once that lands, I could implement a loop to report the errors within `this._delayedFallbackFeatureIds` in the `fallback` method, or the errors could be reported in these two places when we trigger a console.warn. Which would you prefer?

Reply:
Good point. I'm thinking it's easiest if we can consolidate the telemetry reporting in one place, but we can see how it looks once the other PR is merged.

Reply:
Thanks for pointing that out @Snuffleupagus. I opted to create a helper method to avoid `reportTelemetry` code duplication. 

Another option I considered was calling fallback here but passing in an optional bool that would tell us to immediately exit the function following the telemetry collection.

Which solution do you each prefer? @timvandermeij

Reply:
> Another option I considered was calling fallback here but passing in an optional bool that would tell us to immediately exit the function following the telemetry collection.

That honestly sounds worse than the current solution, as far as I'm concerned :-)

(There may be potential for some additional clean-up, since I do think that we're now tracking a bit more state than is probably desirable, but that's perhaps better for a follow-up patch.)